File: all/1653_minimum_deletions_to_make_string_balanced/sol_pairing.py
class Solution:
    def minimumDeletions(self, s: str) -> int:
        minimum_deletions_count = 0
        current_not_dealt_b_count = 0

        for c in s:
            if 'b' == c:
                current_not_dealt_b_count += 1
            elif ('a' == c) and current_not_dealt_b_count:
                current_not_dealt_b_count -= 1
                minimum_deletions_count += 1

        return minimum_deletions_count

File: all/1653_minimum_deletions_to_make_string_balanced/sol_pairing.c
int minimumDeletions(char* s) {
    int minimum_deletions_count = 0;
    int current_not_dealt_b_count = 0;

    for (int i = 0; '\0' != s[i]; ++i) {
        if ('b' == s[i]) {
            ++current_not_dealt_b_count;
        }
        else if ('a' == s[i] && current_not_dealt_b_count) {
            --current_not_dealt_b_count;
            ++minimum_deletions_count;
        }
    }

    return minimum_deletions_count;
}
